fdbserver/storageserver.actor.cpp

```diff
@@ -10534,7 +10534,10 @@ class StorageUpdater {
 			// We can also ignore clearRanges, because they are always accompanied by such a pair of sets with the same
 			// keys
 			startKey = m.param1;
-			decodeServerKeysValue(m.param2, nowAssigned, emptyRange, enablePSM, dataMoveId);
+			DataMoveType dataMoveType = DataMoveType::LOGICAL;
+			decodeServerKeysValue(m.param2, nowAssigned, emptyRange, dataMoveType, dataMoveId);
+			enablePSM = EnablePhysicalShardMove(dataMoveType == DataMoveType::PHYSICAL ||
+			                                    (dataMoveType == DataMoveType::PHYSICAL_EXP && data->isTss()));
```

Comment:
Why do we regard `dataMoveType == DataMoveType::PHYSICAL_EXP && !data->isTss())` as  `!enablePSM`?

Reply:
I understand this now. Thank you for your above explanation.

Reply:
Same as above, `DataMoveType::PHYSICAL_EXP` does not allow SS to do physical shard move, it only allows TSS

Reply:
Yes. Understand. Thanks;